sundials4py: expose optional KLU SUNLinearSolver bindings (#966)
LLNL/sundials: doc/shared/Python/Usage.rst

Other SUNDIALS options can also be accessed in this way. Review
:numref:`Installation.Options` for more information on the available options.

To build sundials4py with KLU support enabled, pass CMake options for KLU (and
SuiteSparse discovery, if needed) through ``CMAKE_ARGS`` when building from
source:

.. code-block:: bash

>    export CMAKE_ARGS="-DSUNDIALS_ENABLE_KLU=ON -DSuiteSparse_ROOT=/path/to/suitesparse"

Reviewer: This is not the correct CMake flag, and "discovery" is not our typical language in our installation documentation.
```suggestion
To build sundials4py with KLU support enabled, pass CMake options for KLU (see :numref:`Installation.Options.KLU`) through ``CMAKE_ARGS`` when building from
source:

.. code-block:: bash

   export CMAKE_ARGS="-DSUNDIALS_ENABLE_KLU=ON -DKLU_ROOT=/path/to/suitesparse/installation"
```

.. note::

   Not all SUNDIALS features are supported by the Python interfaces. In

>    particular, third-party libraries are not yet broadly supported.

Reviewer: I wouldn't consider a single third-party library as being close to "broad" support.
```suggestion
   particular, third-party libraries are only partially supported.
```

information on usage differences, continue to the :ref:`next section
<Python.Usage.Differences>`. Additional examples can be found in the

> ``examples/python`` directory of the :examples:`SUNDIALS GitHub repository

Reviewer 2: ```suggestion
``examples/python`` directory of the :examples:`SUNDIALS GitHub repository.
```

> .

Reviewer 2: ```suggestion
```

information on usage differences, continue to the :ref:`next section
<Python.Usage.Differences>`. Additional examples can be found in the
``examples/python`` directory of the :examples:`SUNDIALS GitHub repository
<python>`, including the KLU example

> ``examples/python/arkode/ark_brusselator1D_klu.py``.

Reviewer: There are now a lot of other examples, so it's unclear why only this one should be highlighted here.  How about this for a revision?
```suggestion
<python>`.  We note the example that leverages the KLU third-party library,
``examples/python/arkode/ark_brusselator1D_klu.py``.
```

Reviewer 2: ```suggestion
.
```

Reviewer 2: I dont think we need to mention it at all.
